### PR TITLE
Add Consolidated view classifier to make view types internal

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public class ConsolidatedMvcViewDocumentClassifierPass : DocumentClassifierPassBase
```
Comment (pranavkm) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:10]: Btw, we could make this internal if it's added by default.

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        public static readonly string MvcViewDocumentKind = "mvc.1.0.view";
+
+        protected override string DocumentKind => MvcViewDocumentKind;
+
+        protected override bool IsMatch(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode) => true;
+            
+
```
Comment (pranavkm) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:18]:
```suggestion

```

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                // Eg. When we try to generate code for an in memory document like default imports.
+                var checksum = Checksum.BytesToString(codeDocument.Source.GetChecksum());
+                @class.ClassName = $"AspNetCore_{checksum}";
+            }
+            else
+            {
+                @class.ClassName = className;
+            }
+            
+
```
Comment (pranavkm) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:48]: extra line

File: src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorConfiguration.cs [hunk 4]
```diff
@@ -13,12 +13,14 @@ public abstract class RazorConfiguration : IEquatable<RazorConfiguration>
...
-            Array.Empty<RazorExtension>());
+            Array.Empty<RazorExtension>(),
+            false);
 
         public static RazorConfiguration Create(
             RazorLanguageVersion languageVersion,
             string configurationName,
-            IEnumerable<RazorExtension> extensions)
+            IEnumerable<RazorExtension> extensions,
+            bool isConsolidatedMvcViews = false)
```
Comment (pranavkm) [inline src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorConfiguration.cs:23]: `useConsolidatedViews`?

**Review by pranavkm — APPROVED** [review]
Looks dope. Have you verified this works e2e?

**Comment by pranavkm** [comment]
Could you bring this up in API review next week?

**Comment by captainsafia** [comment]
@pranavkm Yep.

Screen Shot 2021-03-16 at 11 49 36 AM

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public class ConsolidatedMvcViewDocumentClassifierPass : DocumentClassifierPassBase
```
Comment (JamesNK) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:10]: Should this class be sealed?

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            }
+            else
+            {
+                @class.ClassName = className;
+            }
+            
+            @class.BaseType = "global::Microsoft.AspNetCore.Mvc.Razor.RazorPage<TModel>";
+            @class.Modifiers.Clear();
+            @class.Modifiers.Add("internal");
+            @class.Modifiers.Add("sealed");
```
Comment (pranavkm) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:50]: FYI @benaadams

Comment (benaadams) [reply src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:50]: 😍 Finally! 😁

Comment (benaadams) [reply src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:50]: Will be interested to see if this shows up in benchmarks

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public sealed class ConsolidatedMvcViewDocumentClassifierPass : DocumentClassifierPassBase
+    {
+        public static readonly string MvcViewDocumentKind = "mvc.1.0.view";
```
Comment (rynowak) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:12]: Good to know MVC is still a 1.0 :)

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs (hunk 6 above)
Comment (rynowak) [reply src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:50]: I'm super glad about this tooooo

Comment (benaadams) [reply src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ConsolidatedMvcViewDocumentClassifierPass.cs:50]: Not sure MVC views/pages are currently running in aspnet/benchmarks 😢

**Pull request merged**